heron/spi/src/java/com/twitter/heron/spi/utils/ShellUtils.java

```diff
@@ -82,13 +82,36 @@ public static int runSyncProcess(
   }
 
   /**
+   * Start a daemon thread to read data from "input" to "out".
+   */
+  private static Thread asyncProcessStream(final InputStream input, final StringBuilder out) {
+    Thread thread = new Thread() {
+      @Override
+      public void run() {
+        try {
+          out.append(inputstreamToString(input));
+        } finally {
+          try {
+            input.close();
+          } catch (IOException e) {
+            LOG.log(Level.SEVERE, "Failed to close the input stream", e);
```

Comment:
since we're logging this and moving on without throwing, we should change this to Level.WARN actually.